components/brave_search/renderer/brave_search_js_handler.cc

```diff
@@ -0,0 +1,132 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_search/renderer/brave_search_js_handler.h"
+
+#include <utility>
+
+#include "base/no_destructor.h"
+#include "base/strings/utf_string_conversions.h"
+#include "content/public/renderer/render_frame.h"
+#include "gin/arguments.h"
+#include "gin/function_template.h"
+#include "third_party/blink/public/common/browser_interface_broker_proxy.h"
+#include "third_party/blink/public/web/blink.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "third_party/blink/public/web/web_script_source.h"
+
+namespace brave_search {
+
+BraveSearchJSHandler::BraveSearchJSHandler(content::RenderFrame* render_frame)
+    : render_frame_(render_frame) {}
+
+BraveSearchJSHandler::~BraveSearchJSHandler() = default;
+
+bool BraveSearchJSHandler::EnsureConnected() {
+  if (!brave_search_fallback_.is_bound()) {
+    render_frame_->GetBrowserInterfaceBroker()->GetInterface(
+        brave_search_fallback_.BindNewPipeAndPassReceiver());
+  }
+
+  return brave_search_fallback_.is_bound();
+}
+
+void BraveSearchJSHandler::AddJavaScriptObjectToFrame(
+    v8::Local<v8::Context> context) {
+  v8::Isolate* isolate = blink::MainThreadIsolate();
+  v8::HandleScope handle_scope(isolate);
+  if (context.IsEmpty())
+    return;
+
+  v8::Context::Scope context_scope(context);
+
+  CreateAFallbackObject(isolate, context);
+}
+
+void BraveSearchJSHandler::CreateAFallbackObject(
```

Comment:
why is this a "fallback" object? Also we're not creating anything here, right? Seems like we should combine this with `BindFunctionsToObject` and just call it `BindFunctionsToChrome` ?

Reply:
oh, actually didn't we want this to be `chrome.braveSearch.fetchBackupResults`? @bbondy ?